commons/commons-model-extended-serialization/src/main/kotlin/uk/ac/ebi/extended/serialization/serializers/ExtFileSerializer.kt

```diff
@@ -53,26 +48,13 @@ class ExtFileSerializer : JsonSerializer<ExtFile>() {
         writeStringField(FILE_REL_PATH, file.relPath)
         writeStringField(FILE_FIRE_ID, file.fireId)
         writeObjectField(ATTRIBUTES, file.attributes)
-        writeStringField(EXT_TYPE, ExtType.FireFile.type)
-        writeStringField(FILE_TYPE, FILE_FILE_TYPE)
+        writeStringField(EXT_TYPE, file.extType)
+        writeStringField(FILE_TYPE, file.type.value)
         writeStringField(FILE_MD5, file.md5)
         writeNumberField(FILE_SIZE, file.size)
         writeEndObject()
     }
 
-    private fun JsonGenerator.serializeFireDirectory(file: FireDirectory) {
-        writeStartObject()
-        writeStringField(FILE_NAME, file.fileName)
-        writeStringField(FILE_FILEPATH, file.filePath)
-        writeStringField(FILE_REL_PATH, file.relPath)
-        writeStringField(FILE_FIRE_ID, file.fireId)
-        writeObjectField(ATTRIBUTES, file.attributes)
-        writeStringField(EXT_TYPE, ExtType.FireDirectory.type)
-        writeStringField(FILE_TYPE, FILE_DIR_TYPE)
-        writeStringField(FILE_MD5, file.md5)
-        writeNumberField(FILE_SIZE, file.size)
-        writeEndObject()
-    }
-
-    private fun getType(file: File) = if (FileUtils.isDirectory(file)) FILE_DIR_TYPE else FILE_FILE_TYPE
+    private val FireFile.extType: String
```

Comment:
not sure about this one,  basically I am not sure when is better to use an extension function, and property and or a regular function when is private as this one and others in this PR. 

on one hand sintax is nice but in the other hand an private extension property is more complex thant a simple function (well know in every single programing language). IMO the nice sintax (used only in one or two places) does not pay off the extra complexity but not 100% sure.

Reply:
Well, in this case, I didn't use a function since it would be a "virtual" property from the FireFile. That being said, look at how cute it looks 🤩 

```
private val FireFile.extType: String
        get() = when(type) {
            FILE -> ExtType.FireFile.type
            DIR -> ExtType.FireDirectory.type
        }
```

Reply:
Never mind, this isn't actually necessary since we really should have the file type telling us if it's either a file or directory and the ext file type telling us whether it's on NFS or FIRE. Please check my latest changes here.